### PR TITLE
Update migrated SDKs and fix Razor server

### PR DESCRIPTION
**Pull request by captainsafia**
- Migrate https://github.com/dotnet/aspnetcore/pull/29293 and https://github.com/dotnet/aspnetcore/pull/28714 into Blazor WASM SDK
- Fix ArgumentEscaper in ServerConnection to improve perf
- Remove unneeded `Condition="'$(IsCrossTargetingBuild)' != 'true'"` check in Razor SDK targets
- Bring back all the Blazor WASM tests now that the underlying perf issue is fixed
- Bring in changes from https://github.com/dotnet/aspnetcore/pull/27974

### CODE REVIEW

File: src/RazorSdk/Sdk/Sdk.targets [hunk 1]
```diff
@@ -14,7 +14,7 @@ Copyright (c) .NET Foundation. All rights reserved.
   <Import Sdk="Microsoft.NET.Sdk" Project="Sdk.targets" Condition="'$(_RazorSdkImportsMicrosoftNetSdk)' == 'true'" />
 
   <PropertyGroup Condition="'$(RazorSdkCurrentVersionTargets)' == ''">
-    <RazorSdkCurrentVersionTargets Condition="'$(IsCrossTargetingBuild)' != 'true'">$(MSBuildThisFileDirectory)..\build\netstandard2.0\Sdk.Razor.CurrentVersion.targets</RazorSdkCurrentVersionTargets>
+    <RazorSdkCurrentVersionTargets>$(MSBuildThisFileDirectory)..\build\netstandard2.0\Sdk.Razor.CurrentVersion.targets</RazorSdkCurrentVersionTargets>
```
Comment (pranavkm) [inline src/RazorSdk/Sdk/Sdk.targets:17]: hmm?

File: src/RazorSdk/Tasks/Microsoft.NET.Sdk.Razor.Tasks.csproj [hunk 2]
```diff
@@ -71,6 +71,9 @@
 
   <ItemGroup>
     <Compile Include="**\*.cs" />
+    <Compile Include="..\Tool\CommandLine\*.cs">
```
Comment (pranavkm) [inline src/RazorSdk/Tasks/Microsoft.NET.Sdk.Razor.Tasks.csproj:74]: Do you need all the files from this directory?

File: src/RazorSdk/Sdk/Sdk.targets (hunk 1 above)
Comment (captainsafia) [reply src/RazorSdk/Sdk/Sdk.targets:17]: When we introduced this PR, we removed the cross-targeting references so I don't think we need to this anymore. We'll always import these targets, correct?

File: src/RazorSdk/Tasks/Microsoft.NET.Sdk.Razor.Tasks.csproj (hunk 2 above)
Comment (captainsafia) [reply src/RazorSdk/Tasks/Microsoft.NET.Sdk.Razor.Tasks.csproj:74]: Nope. Just the `ArgumentEscaper`. I can be more specific here.

File: src/Assets/TestProjects/BlazorMultiApp/BlazorMultipleApps.Server/BlazorMultipleApps.Server.csproj [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  <ItemGroup>
+    <ProjectReference Include="..\BlazorMultipleApps.SecondClient\BlazorMultipleApps.SecondClient.csproj" />
+    <ProjectReference Include="..\BlazorMultipleApps.FirstClient\BlazorMultipleApps.FirstClient.csproj" />
+    <ProjectReference Include="..\BlazorMultipleApps.Shared\BlazorMultipleApps.Shared.csproj" />
+  </ItemGroup>
+
+  <!-- Enabling compression increases build time. We want to avoid this for tests so we
+  disable it here. This doesn't affect any test assertions. -->
+  <PropertyGroup>
+    <_BlazorBrotliCompressionLevel>NoCompression</_BlazorBrotliCompressionLevel>
```
Comment (pranavkm) [inline src/Assets/TestProjects/BlazorMultiApp/BlazorMultipleApps.Server/BlazorMultipleApps.Server.csproj:16]: You'll need these in the two client projects.

File: src/Assets/TestProjects/BlazorMultiApp/BlazorMultipleApps.SecondClient/BlazorMultipleApps.SecondClient.csproj [hunk 4]
```diff
@@ -0,0 +1,17 @@
+﻿<Project Sdk="Microsoft.NET.Sdk.BlazorWebAssembly">
+
+  <PropertyGroup>
+    <TargetFramework>net5.0</TargetFramework>
+    <StaticWebAssetBasePath>SecondApp</StaticWebAssetBasePath>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly" Version="5.0.0" />
+    <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly.DevServer" Version="5.0.0" PrivateAssets="all" />
```
Comment (pranavkm) [inline src/Assets/TestProjects/BlazorMultiApp/BlazorMultipleApps.SecondClient/BlazorMultipleApps.SecondClient.csproj:10]: Feel free to remove this

File: src/Assets/TestProjects/BlazorMultiApp/BlazorMultipleApps.FirstClient/BlazorMultipleApps.FirstClient.csproj [hunk 5]
```diff
@@ -0,0 +1,18 @@
+<Project Sdk="Microsoft.NET.Sdk.BlazorWebAssembly">
+
+  <PropertyGroup>
+    <TargetFramework>net5.0</TargetFramework>
+    <StaticWebAssetBasePath>FirstApp</StaticWebAssetBasePath>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly" Version="5.0.0" />
+    <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly.DevServer" Version="5.0.0" PrivateAssets="all" />
```
Comment (pranavkm) [inline src/Assets/TestProjects/BlazorMultiApp/BlazorMultipleApps.FirstClient/BlazorMultipleApps.FirstClient.csproj:10]: Also this. Feel free to yank it out.

**Review by pranavkm — APPROVED** [review]

**Pull request merged**